libs/sdk/skyux-eslint/src/rules/template/prefer-label-text.ts

```diff
@@ -56,6 +63,105 @@ const SELECTORS_WITH_LABEL_COMPONENTS = COMPONENTS_WITH_LABEL_TEXT.map(
   (c) => c.selector,
 ).join('|');
 
+const FORM_CONTROL_CLASS = 'sky-form-control';
+
+function getAttributeByName(
+  el: TmplAstElement,
+  attrName: string,
+): TmplAstTextAttribute | undefined {
+  for (const attribute of el.attributes) {
+    if (attribute.name === attrName) {
+      return attribute;
+    }
+  }
+
+  return;
+}
+
+/**
+ * Removes the "sky-form-control" CSS class from <sky-input-box /> input elements
+ * to satisfy the input box's input directive selector.
+ * See: https://github.com/blackbaud/skyux/blob/040e461a50cb3d08ff8f7332dba350b7e97c5fd8/libs/components/forms/src/lib/modules/input-box/input-box-control.directive.ts#L11
+ */
+function removeFormControlClass(
+  fixer: RuleFixer,
+  inputEl: TmplAstElement,
+): RuleFix[] {
+  const fixes: RuleFix[] = [];
+
+  const classAttr = getAttributeByName(inputEl, 'class');
+
+  if (classAttr?.value.includes(FORM_CONTROL_CLASS)) {
+    const classnames = classAttr.value;
+
+    // Remove "class" attribute if it only contains the form control class.
+    if (classnames === FORM_CONTROL_CLASS) {
+      fixes.push(
+        fixer.removeRange([
+          classAttr.sourceSpan.start.offset,
+          classAttr.sourceSpan.end.offset,
+        ]),
+      );
+    } else {
+      const index = classnames.indexOf('sky-form-control');
+      const classLength = FORM_CONTROL_CLASS.length;
+
+      /* istanbul ignore else: safety check */
+      if (classAttr.valueSpan) {
+        let classStart = classAttr.valueSpan.start.offset + index;
+        let classEnd = classStart + classLength;
+
+        // Account for multiple classes in the "class" attribute, and remove extra
+        // space characters.
+        if (classnames.at(index - 1) === ' ') {
+          classStart--;
+        } else if (classnames.at(index + classLength) === ' ') {
+          classEnd++;
+        }
+
+        fixes.push(fixer.removeRange([classStart, classEnd]));
+      }
+    }
+  }
+
+  return fixes;
+}
+
+/**
+ * Removes the "skyId" directive from <sky-input-box /> input elements to
+ * satisfy the input box's input directive selector.
+ * See: https://github.com/blackbaud/skyux/blob/040e461a50cb3d08ff8f7332dba350b7e97c5fd8/libs/components/forms/src/lib/modules/input-box/input-box-control.directive.ts#L11
+ */
+function removeSkyIdAttributes(
+  fixer: RuleFixer,
+  inputEl: TmplAstElement,
+): RuleFix[] {
+  const fixers: RuleFix[] = [];
+  const skyIdAttr = getAttributeByName(inputEl, 'skyId');
```

Comment:
You may want to check to see if the `skyId` for the input is being used anywhere else in the template, like in an element's `aria-label` attribute.

Reply:
Good callout. We decided to abort the rule altogether if `skyId` found since it usually means the user wants to do "hard" mode.